test/integration/emitter_singleton_test.ts

```diff
@@ -0,0 +1,56 @@
+import "mocha";
+
+import { emitterFor, HTTP, Mode, Message, Emitter } from "../../src";
+
+import { fixture, assertStructured } from "./emitter_factory_test";
+
+import { rejects, doesNotReject } from "assert";
+
+describe("Emitter Singleton", () => {
+  it("emit", async () => {
+    let msg: Message | unknown = undefined;
+    const fn = async (message: Message) => {
+      msg = message;
+    };
+    const emitter = emitterFor(fn, { binding: HTTP, mode: Mode.STRUCTURED });
+    Emitter.on("cloudevent", emitter);
+    fixture.emit();
+    while (msg === undefined) {
```

Comment:
A while loop like this, waiting for a response, is really bad practice. I'd imagine an await pattern after emitting.

Reply:
I agree overall but...
The NodeJS eventing system is not designed with Promise https://nodejs.org/api/events.html

Reply:
I'm sorry, but there are better methods to test an asynchronous pattern than a while undefined loop. A while loop like this is blocking I/O.

If we did want to go this route, in Node, it's common to not write `=== false` or `=== undefined`, rather use  `!`. If we can, I'd `await` the event.

Reply:
You mean this one: https://nodejs.org/api/events.html#events_events_on_emitter_eventname_options because it is not available on Node 10 which is one of the target

I do not know other way to get a promise on an event, how would you do it?

Reply:
Also, can you iterate on the blocking I/O as this loop include the await of timeout so I wonder why/how it is blocking the I/O.
Thanks for clarifying so I can get a better understanding of NodeJS

Reply:
Right now the `on` function implemented does not return a `Promise`. Promises are a good way to have async processing in general.

You could use the Events emitter.on pattern too. Whatever the choice, I would not have a while loop like this, or would imagine exponential backoff if required.

Reply:
The NodeJS pattern is available only on Node v12.16.0 so cannot be used.
I just updated the PR with a custom Promise that will resolve by the event itself mimicking a bit the pattern